00-framework-01-maturity.md

```diff
@@ -0,0 +1,118 @@
+# libp2p specification framework – the five-stage maturity process
+
+> Author: @raulk
+
+> Revision: r0, 2019-05-21
+
+## Prelude
+
+This document establishes a framework to foster rapid and incremental libp2p
+specification development, aiming to reduce the barrier to entry for new
+ideas, increasing the throughput of breakthrough creative proposals, promoting
+their evolution and adoption within the community, while maximising consensus
+around a common lifecycle and criteria for progression across stages.
+
+We propose a five-stage process to classify the maturity of a specification in
+terms of completeness, demonstrability of implementation, community
+acceptance, and level of technical detail. It is somewhat inspired in the W3C
+Process [0].
+
+  1. `Working Draft`
+  2. `Candidate Recommendation`
+  3. `Recommendation`
+  4. `Deprecated`
+  5. `Abandoned Draft`
+
+## Stage 1: Working draft
+
+The specification of the system, process, protocol or item is under active
+development. This stage is lightweight and mostly self-directed by the author.
+We aim to reduce the barrier to entry, and it's designed to allow for
+iterative experimentation, discovery and pivoting.
+
+We don't enforce a hard template in an attempt to enhance author
+expressability and creativity.
+
+We enter this stage by posting an `Initial Working Draft` that covers:
+  * context: what is the current situation or a brief overview of the
+    environment the specification targets.
+  * motivation: why this specification is relevant, and how it advances the
+    status quo.
+  * scope and rationale: what areas of the technical system the specification
+    impacts.
+  * goals: what we expect to achieve (positively and negatively) as a result
+    of implementing the specification.
+  * expected feature set: a summary/enumeration of features the spec provides.
+  * tentative technical directions: how are we planning to materialise the
+    specification in terms of system design.
+
+The `Initial Working Draft` shall be reviewed by the community in no more than 3
```

Comment:
that's too strict, let's say a week or two.

Reply:
Ok, I'll bump this up to 5 days (1 working week). Two weeks is too much to ingest an idea for incubation.

Reply:
I am concerned with us having the bandwidth to review the spec, and a proper review might take time.